stdlib/public/Concurrency/ExecutorAssertions.swift

```diff
@@ -343,13 +343,24 @@ extension Actor {
   /// - Throws: rethrows the `Error` thrown by the operation if it threw
   @available(SwiftStdlib 5.1, *)
   #if !$Embedded
-  @backDeployed(before: SwiftStdlib 5.9)
+  @_alwaysEmitIntoClient
   #endif
   @_unavailableFromAsync(message: "express the closure as an explicit function declared on the specified 'actor' instead")
   @_unavailableInEmbedded
-  public nonisolated func assumeIsolated<T>(
+  public nonisolated func assumeIsolated<T : Sendable>(
       _ operation: (isolated Self) throws -> T,
       file: StaticString = #fileID, line: UInt = #line
+  ) rethrows -> T {
+    try assumeIsolatedOutOfLine(operation, file, line)
```

Comment:
This doesn't look it will work when back deployed. `assumeIsolatedOutOfLine()` corresponds to an ABI entry point that was introduced in Swift 5.9 right? Then the body of an `@_alwaysEmitIntoClient` function that is available as far back as runtimes corresponding to the Swift 5.1 era cannot call it. If you annotate `assumeIsolatedOutOfLine` with an accurate `@available` attribute the availability checker would diagnose this.

Reply:
I think you want to invert this. The body of the function should go in the new `@_alwaysEmitIntoClient` function and the ABI shim should call the new function. The shim still needs accurate availability, though.